src/common/line_entry.rs

```diff
@@ -72,6 +72,52 @@ impl LineEntry {
         }
         comment::parse(self.raw_string.as_str())
     }
+
+    #[allow(dead_code)]
+    pub fn get_substitution_keys(&self) -> Vec<&str> {
+        let mut keys = Vec::new();
+
+        let mut value = match self.get_value().map(str::trim) {
+            Some(value) if !value.starts_with('\'') => value,
+            _ => return keys,
+        };
+
+        let escaped =
+            |prefix: &str| prefix.chars().rev().take_while(|ch| *ch == '\\').count() % 2 == 1;
+
+        if value.starts_with('\"') {
+            if value.ends_with('\"') && !escaped(&value[..value.len() - 1]) {
+                value = &value[1..value.len() - 1]
+            } else {
+                return keys;
+            }
+        }
+
+        while let Some(index) = value.find('$') {
+            let prefix = &value[..index];
+            let raw_key = &value[index + 1..];
+
+            if escaped(prefix) {
+                value = &raw_key;
+            } else {
+                let (key, rest) = match raw_key.chars().next() {
+                    Some('{') => match raw_key.find('}') {
+                        Some(index) => (&raw_key[1..index], &raw_key[index + 1..]),
+                        None => return keys,
+                    },
+                    Some(_) => match raw_key.find(|c: char| !c.is_ascii_alphanumeric() && c != '_')
+                    {
+                        Some(index) => raw_key.split_at(index),
+                        None => (raw_key, ""),
+                    },
+                    None => return keys,
+                };
+                keys.push(key);
```

Comment:
What do you think of a more functional style here? 🤔  
Personally, I think this option is clearer 🙂
```suggestion
                let (key, rest) = raw_key
                    .strip_prefix('{')
                    .and_then(|raw_key| raw_key.find('}').map(|i| raw_key.split_at(i)))
                    .or_else(|| {
                        raw_key
                            .find(|c: char| !c.is_ascii_alphanumeric() && c != '_')
                            .map(|i| raw_key.split_at(i))
                    })
                    .unwrap_or((raw_key, ""));

                if !key.is_empty() {
                    keys.push(key);
                }
```

Reply:
Great. Done. But to be precise for equal behaviour we [should return](https://github.com/dotenv-linter/dotenv-linter/pull/391/commits/0b5826f768088e3afdf7ba233b404838d092de22#diff-ddfe155afc579aaf9bf97e6f367bd45f7cbe5a69d01666f5d8f19330218a438dR115) `keys` in case of empty `key`. Added [some test](https://github.com/dotenv-linter/dotenv-linter/pull/391/commits/0b5826f768088e3afdf7ba233b404838d092de22#diff-ddfe155afc579aaf9bf97e6f367bd45f7cbe5a69d01666f5d8f19330218a438dR405) for this case. Is it ok? How do we decide?